Update default template with UG links and useful information

lhw-1 opened the pull request:
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [X] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

<!--
  If this pull request is addressing an issue, link to the issue: #xxx

  If this pull request completely addresses an issue, use one of the closing keywords: "Fixes #xxx" or "Resolves #xxx"
  https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

  Otherwise, elaborate further on the rationale of this pull request as needed
-->

**Overview of changes:**

Addresses a few of the main outcomes of the discussions in #2214, and updates the `default` template to include links to the main sections of the User Guide. Also possibly addresses #519.

Here is a preview of what it currently looks like:

![image](https://user-images.githubusercontent.com/25716529/226122741-30a654b9-e36a-46df-b23d-03425279f376.png)

I've kept the new look to be as simple as possible, and have modelled it based partially on the MarkBind [User Guide - Authoring Contents](https://markbind.org/userGuide/authoringContents.html) section and partially on the [Gatsby initial site](https://egghead.io/lessons/gatsby-use-npm-init-gatsby-to-create-an-initial-gatsby-site).

This is just a possible direction we could take the default template towards - if there are any suggestions to improve this current look, or if there are any concerns / issues with this, please highlight as needed!

**Anything you'd like to highlight/discuss:**

We should keep in mind to keep the template as simple as possible while including the main features that a user using `markbind init --convert --template default` would want to have in their site. (E.g. siteNav and pageNav are essential for this - any other suggestions are welcome.)

**Testing instructions:**

Run `markbind init` in a new directory to initialize a site built using the `default` template. 

Alternatively, you can view the deployed version of the template [here](https://lhw-1.github.io/markbind-2225/).

<!--
  Any special testing instructions **not including** our automated tests.
-->

**Proposed commit message: (wrap lines at 72 characters)**
```
Update default template with UG links and useful information

The current default template contains filler content and
component usage that is not well explained.

This could cause confusion to users who are starting with the
default template from scratch for their site.

Updating the default template allows us to help guide the newer
users, and since the content is overriden when being converted,
will not affect users who are converting pre-existing documents.

Let's add links to our User Guide, and re-organize the information
given to the users in a more succinct manner, while also
showcasing MarkBind components.
```
<!--
  See this link for more info on how to write a good commit message:
  https://se-education.org/guides/conventions/git.html

  As best as possible, write a succinct commit title in 50 characters

|---------This is the width of 50 chars----------|
|-----------This is the width of 72 chars for your reference-----------|
-->

---

**Checklist:** :ballot_box_with_check:

<!-- Leave non-applicable items unchecked -->

- [X] Updated the documentation for feature additions and enhancements
- [X] Added tests for bug fixes or features
- [X] Linked all related issues
- [X] No unrelated changes <!-- It's tempting, but increases the reviewer's work, and really pollutes the commit history =( -->

<!--
  We'll try our best to get to your PR within a week.
  If we haven't gotten to it then, or if your pull request resolves an urgent item, feel free to give us a ping!
-->

ong6 commented on packages/core/template/default/index.md:
```diff
@@ -9,124 +9,45 @@
+<hr>
 
-**A block quote:**
+<h5><strong>User Guide: Authoring Contents</strong></h5>
```
Instead of using pure html in the whole file, can we use a mix of html and markdown instead? This might be easier for first time users and is also one of our selling points. 

I think the headings should at least be done in markdown instead of html

lhw-1 replied:
That's a good idea! I've changed the headings to markdown in [d5317d0](https://github.com/MarkBind/markbind/pull/2225/commits/d5317d095a7a908d8a582e36e54a7751e4889a53).

jovyntls commented on packages/core/template/default/index.md:
```diff
@@ -9,124 +9,45 @@
   <div class="container">
-    <h1 class="display-4 no-index">Landing Page Title</h1>
-    <p class="lead">A tagline can go here</p>
+    <h1 class="display-4 no-index">Great!<br>You just initialized a MarkBind site.</h1>
```
Minor nit!
```suggestion
    <h1 class="display-4 no-index">Great!<br>You've just initialized a MarkBind site.</h1>
```

jovyntls reviewed (commented):
Template mostly LGTM! I'm not a fan of the empty pagenav on the right (it seems a bit weird to have a pagenav that has a "Chapters of this page" header but no contents) - could we add some unobtrusive headers to populate the pagenav? I think it's still worth showcasing that the pagenav exists since it's one of the key features of the default template.

jovyntls commented on docs/userGuide/templates.md:
```diff
@@ -20,5 +20,5 @@ markbind init --template minimal
-Default | `default`    | Default template if `--template` is unspecified. Filled with MarkBind features to guide you to author better content. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-typical-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
-Minimal | `minimal` | Minimalistic template that gets you started quickly | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-minimal-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
\ No newline at end of file
+Default | `default`    | Default template if `--template` is unspecified. Includes links to the main sections of the MarkBind User Guide. | <a href="https://app.netlify.com/start/deploy?repository=https://github.com/MarkBind/init-typical-netlify"><img src="https://www.netlify.com/img/deploy/button.svg" /></a>
```
I think we might not need to mention that there are links to the UG here, since it's not really a feature of the template. Maybe we can relate this to the template being a quick start option?

lhw-1 replied:
Good point! Would something like "Includes core features such as site and page navigation for a more convenient quick start." be better?

lhw-1 commented:
> Template mostly LGTM! I'm not a fan of the empty pagenav on the right (it seems a bit weird to have a pagenav that has a "Chapters of this page" header but no contents) - could we add some unobtrusive headers to populate the pagenav? I think it's still worth showcasing that the pagenav exists since it's one of the key features of the default template.

That's a good point - I'll include more sections from the main MarkBind page to make the site longer (and thus have functional page nav) :+1:

tlylt commented:
@lhw-1 maybe set up a site, deploy it, and add the link to the description? Easier to preview that way?

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -9,124 +9,45 @@
+##### **User Guide: Working with Sites**
 
-A <tooltip content=":exclamation: some **important explanation**" placement="top" trigger="hover">tooltip</tooltip>, a <trigger for="modal:modalinfo" trigger="click">modal</trigger>, a <a href="https://markbind.org/">link</a>, a <span class="badge bg-danger">badge</span>, another <span class="badge bg-warning text-dark">badge</span>.
+<blockquote>
```
Any problem with `>` instead of `<blockquote>` ? I would prefer Markdown over direct HTML. Same for other content in this page

lhw-1 replied:
Absolutely no problem! I simply copied what was being done in the [Authoring Contents file](https://github.com/MarkBind/markbind/blob/master/docs/userGuide/authoringContents.md). 

Markdown definitely does make it cleaner - in [44f4454](https://github.com/MarkBind/markbind/pull/2225/commits/44f44542396dd4ab489d92fbe067fcb3498f174a), I've changed the `<blockquote>` to `>`, replaced `<hr>` with `---`, and removed the `<span>`. For now, I kept the blue heading container (to keep the style) and the `<a>` tags (to make use of `target="_blank"`).

tlylt reviewed (changes requested):
Thanks @ lhw-1, this revamp is great. Some nits as mentioned.

As much as possible, I think we should use MarkBind features on this page to present the appropriate information. Not suggesting that we go back to the current site where we include different versions of the panel state to showcase the style of the header when expanded or minimized, but use appropriate components to display information. Panels, box, tooltip, etc are all possible.

lhw-1 commented:
> I'll look at the main page and other relevant sections once more, and see what we can add to the template without bloating it too much. Ideally, it should be long enough to make the page navigation a useful addition (currently it's too short for the page nav to be useful), while making use of a good number of MarkBind components, but not so much to the point of bloating it.

I have updated the template to include more content (to make our pageNav more useful & to show off some of our components). You can check out the new version at the deployed site [here](https://lhw-1.github.io/markbind-2225/).

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+---
 
-**A block quote:**
+## What did I just do?
```
```suggestion
## What did you just do?
```

lhw-1 replied:
Fixed in [945ea73](https://github.com/MarkBind/markbind/pull/2225/commits/945ea738078da9000f4299903da634c4db9dee5e)!

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
-value2   | y        | 20       | :far-thumbs-down:
+## Guide to MarkBind
 
+To see the capability of MarkBind in action, feel free to take a look at some of the websites built using MarkBind on our <a href="https://markbind.org/userGuide/gettingStarted.html" target="_blank">Showcase</a> page. 
```
wrong link

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
   layout: default.md
-  pageNav: 2
+  pageNav: 4
   pageNavTitle: "Chapters of This Page"
```
perhaps a more appropriate nav title here

lhw-1 replied:
Fixed in [945ea73](https://github.com/MarkBind/markbind/pull/2225/commits/945ea738078da9000f4299903da634c4db9dee5e)!

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+> Learn about the variety of syntax schemes, formats, and custom MarkBind components that you can use in your MarkBind site.
 
-**Some boxes:**
+More info in: _<a href="https://markbind.org/userGuide/authoringContents.html" target="_blank">User Guide → Authoring Contents</a>_
```
This link and line 65 is the same, is this a mistake?

lhw-1 replied:
Fixed in [945ea73](https://github.com/MarkBind/markbind/pull/2225/commits/945ea738078da9000f4299903da634c4db9dee5e)!

tlylt reviewed (changes requested):
Thanks @lhw-1, some nits. I think it's almost there.

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
-  pageNav: 2
-  pageNavTitle: "Chapters of This Page"
+  pageNav: 4
+  pageNavTitle: "Placeholder Topic Pages"
```
```suggestion
  pageNavTitle: "Topics"
```

lhw-1 replied:
[fab85db](https://github.com/MarkBind/markbind/pull/2225/commits/fab85db23ed5ba23eacde0a78dc99c45d63d69c7)

tlylt commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+## What did you just do?
 
-> Some text some text some text some text some text some text some text. Some text some text some text some text some text some text some text. Some text some text some text some text some text some text some text some text some text some text some text some text some text some text. Some text some text some text some text some text some text. Some text some text some text some text some text som ...
+You have just initialized a _default_ MarkBind site, which is equipped with a set of core features of a MarkBind site like Site and Page Navigation, as well as some links to our User Guide, to help you get started more conveniently!
```
```suggestion
You have just initialized a _default_ MarkBind site! It is equipped with a set of core features, including site and page navigation. Additionally, we have included some handy links to our User Guide to help you get started quickly and easily.
```

lhw-1 replied:
Thanks for the suggestion! Updated in [fab85db](https://github.com/MarkBind/markbind/pull/2225/commits/fab85db23ed5ba23eacde0a78dc99c45d63d69c7).

tlylt reviewed (approved):
Some nits, otherwise LGTM for me

yucheng11122017 commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+If you were intending to convert an existing GitHub wiki or a docs folder into MarkBind, use the "--convert" flag instead. See <a href="https://markbind.org/userGuide/markBindInTheProjectWorkflow.html#converting-existing-project-documentation-wiki" target="_blank">User Guide: MarkBind in the Project Workflow</a> for more information.
 
-**A `code` example:**
+If you want to start with a <tooltip content="i.e. without any content">_minimal_</tooltip> template instead, use the "--template" flag to initialize a minimal site instead of the default. See <a href="https://markbind.org/userGuide/templates.html" target="_blank">User Guide: Templates</a> for more information.
```
```suggestion
If you want to start with a <tooltip content="i.e. without any content">_minimal_</tooltip> template instead, use the "--template" flag with "minimal" option to initialize a minimal site instead of the default. See <a href="https://markbind.org/userGuide/templates.html" target="_blank">User Guide: Templates</a> for more information.
```

lhw-1 replied:
Thanks for the suggestion! Updated in [fab85db](https://github.com/MarkBind/markbind/pull/2225/commits/fab85db23ed5ba23eacde0a78dc99c45d63d69c7).

yucheng11122017 commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
-<modal header="Modal Title" id="modal:modalinfo">
-Some text some text some text some text some text some text some text. Some text some text some text some text some text some text some text. Some text some text some text some text some text some text some text some text some text some text some text some text some text some text. Some text some text some text some text some text some text. Some text some text some text some text some text some  ...
-</modal>
+This _default_ site comes pre-configured with the core <a href="https://markbind.org/userGuide/components/navigation.html#navigation-components" target="_blank">Navigation components</a>: a <tooltip content="Site Navigation">**siteNav**</tooltip>, a <tooltip content="Page Navigation">**pageNav**</tooltip>, a <tooltip content="Navigation Bar">**NavBar**</tooltip>, and a **Search Bar**</tooltip>. T ...
```
```suggestion
This _default_ site comes pre-configured with the core <a href="https://markbind.org/userGuide/components/navigation.html#navigation-components" target="_blank">Navigation components</a>: a <tooltip content="Site Navigation">**siteNav**</tooltip>, a <tooltip content="Page Navigation">**pageNav**</tooltip>, a <tooltip content="Navigation Bar">**NavBar**</tooltip>, and a **Search Bar**. To help you get started with the **siteNav**, we have included <tooltip content="Topic 1, Topic 2, Topic 3, Topic 3a, Topic 3b">five dummy placeholder pages</tooltip>. The **NavBar** also comes with a placeholder slot for your custom Logo.
```

lhw-1 replied:
Thanks for the find! I didn't notice that :sweat_smile: 

Updated in [fab85db](https://github.com/MarkBind/markbind/pull/2225/commits/fab85db23ed5ba23eacde0a78dc99c45d63d69c7).

yucheng11122017 reviewed (changes requested):
Thanks for the work @lhw-1! This seems to be a much better landing page than the current 👍 
Some minor nits on my end!

lhw-1 commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+##### **User Guide: Working with Sites**
 
-# Heading 3
+> Learn how to modify site properties, apply themes, and enable / disable plugins for your MarkBind site.
```
Thanks for the suggestion! Updated in [fab85db](https://github.com/MarkBind/markbind/pull/2225/commits/fab85db23ed5ba23eacde0a78dc99c45d63d69c7).

jovyntls commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
-  1. item b1
-  1. item b2
-  1. item b3
+If you were intending to convert an existing GitHub wiki or a docs folder into MarkBind, use the "--convert" flag instead. See <a href="https://markbind.org/userGuide/markBindInTheProjectWorkflow.html#converting-existing-project-documentation-wiki" target="_blank">User Guide: MarkBind in the Project Workflow</a> for more information.
```
```suggestion
If you were intending to convert an existing GitHub wiki or a docs folder into MarkBind, use the `--convert` flag instead. See <a href="https://markbind.org/userGuide/markBindInTheProjectWorkflow.html#converting-existing-project-documentation-wiki" target="_blank">User Guide: MarkBind in the Project Workflow</a> for more information.
```
Might be better to use the inline code format? Same for the --template below

lhw-1 replied:
Thanks for the suggestion! Updated in [eface18](https://github.com/MarkBind/markbind/pull/2225/commits/eface184bfc22fc7a9db7f55e82882f47f018d5e).

jovyntls commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
-    </tab>
-  </tab-group>
-</tabs>
+<panel header="**Good starting points in our User Guide:**" expanded no-close>
```
```suggestion
<panel header="**Good starting points in our User Guide**" expanded no-close>
```

lhw-1 replied:
Thanks for the suggestion! Updated in [eface18](https://github.com/MarkBind/markbind/pull/2225/commits/eface184bfc22fc7a9db7f55e82882f47f018d5e).

jovyntls commented on packages/core/template/default/index.md:
```diff
@@ -1,132 +1,75 @@
+
+More info in: _<a href="https://markbind.org/userGuide/fullSyntaxReference.html" target="_blank">User Guide → Full Syntax Reference</a>_
+
+</panel>
```
Can we add some line breaks after the panel here so that the panel isn't tight on the bottom of the page?
<img width="751" alt="Screenshot 2023-03-29 at 4 51 45 PM" src="https://user-images.githubusercontent.com/61113575/228486202-6cce6cf0-77e1-4a43-8269-45d3e625bdec.png">

lhw-1 replied:
Thanks for the suggestion! Updated in [eface18](https://github.com/MarkBind/markbind/pull/2225/commits/eface184bfc22fc7a9db7f55e82882f47f018d5e).

jovyntls reviewed (commented):
LGTM, really minor nits!

Some smaller thoughts:
- "Placeholder Topic Pages" is a bit of a strange title for the page nav. I'm thinking "On this page" might be better? 
- The "What did you just do?" header sounds a bit negative to me, a bit like "What have you done?!?!" 😅 I think this is really minor so I'm fine with leaving it as it is. Other suggestions: "What's going on?", or no header at all

lhw-1 commented:
> Some smaller thoughts:
> 
> * "Placeholder Topic Pages" is a bit of a strange title for the page nav. I'm thinking "On this page" might be better?
> 
> * The "What did you just do?" header sounds a bit negative to me, a bit like "What have you done?!?!" sweat_smile I think this is really minor so I'm fine with leaving it as it is. Other suggestions: "What's going on?", or no header at all

For the first point, @tlylt pointed it out earlier, so I changed it to "Topics"!

As for the second point, that's a good point - I've changed it to "What just happened?" :sweat_smile:

jovyntls reviewed (approved):
LGTM! 🎊

yucheng11122017 commented:
Hi @lhw-1, I'm getting some errors when trying to use the page nav at the side. It doesn't move to the correct header which is really odd... Not sure if its something wrong with the deployment or is it a local thing as well?

tlylt commented:
> Hi @lhw-1, I'm getting some errors when trying to use the page nav at the side. It doesn't move to the correct header which is really odd... Not sure if its something wrong with the deployment or is it a local thing as well?
> 
> 

@lhw-1 can you try building and deploy a version with the latest web built. It could be due to the recent updates to anchor scroll.

yucheng11122017 reviewed (approved):
LGTM! Thanks a lot @lhw-1
